server/webapp/WEB-INF/rails/webpack/models/mixins/errors_consumer.ts

```diff
@@ -0,0 +1,196 @@
+import * as stream from "mithril/stream";
+import {Errors} from "models/mixins/errors";
+import * as s from "underscore.string";
+
+/**
+ * This class can either be used as a base class or a mixin. Currently, this is the base class
+ * for `ValidatableMixin`.
+ *
+ * The concepts encapsulated here are error storage and consuming errors from a server API
+ * response. This is important for any model handling CRUD via the API as there may be errors
+ * that only the server can return because they cannot be validated on the client side. This
+ * allows one to bind server errors back to the respective JS model fields so that the user
+ * can be made aware of errors that prevent the success of user-initiated operations.
+ */
+export class ErrorsConsumer implements ErrorsConsumer {
```

Comment:
Is the `implements` needed? Is it some weird compiler trick?

Reply:
Hmm, that’s a refactoring mistake, though syntactically correct. I will rename the class so it is distinct from the interface. I know that typescript classes are also interfaces but I don’t consider it a good practice to have the concrete class be the same as the interface because it partly removes the benefit of using interfaces, IMO. I’ll fix this, then merge. Thanks for looking at it :)